### PR TITLE
feat(EMS-2777):  No PDF - Policy - Loss Payee - International financial details - Address validation

### PR DESCRIPTION
**Pull request by ttbarnes**
## Introduction :pencil2:
This PR updates the "International financial details" form to validate the address form field.

As part of this PR, a new DRY cypress assertion funtion and UI validation rule has been added to validate and assert financial address fields.

This can later be adapated to be used in "full address" (non-financial) E2E assertions.

## Resolution :heavy_check_mark:
- Create E2E test for "International financial details" form validation.
- Update "UK financial details" form validation E2E test to call `financialAddressFieldValidation`.
- Create new DRY cypress assertion function, `financialAddressFieldValidation`
- Create new UI validation function/rules for "International financial details"  POST controller.
- Update "International financial details" POST controller to call `generateValidationErrors`.
- Create new DRY UI validation rule, `financialAddressRule`.

## Miscellaneous :heavy_plus_sign:
- Update error messages to consume the `MAXIMUM_CHARACTERS.FULL_ADDRESS` constant.
- Fix some typos/test descriptions.
- Fix a test file location/parent directory.
- Minor documentation/function name improvements.

### CODE REVIEW

File: e2e-tests/insurance/cypress/e2e/journeys/policy/loss-payee-financial-details-uk/validation/financial-address-validation.spec.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+const {
+  ROOT: INSURANCE_ROOT,
+  POLICY: {
+    LOSS_PAYEE_FINANCIAL_DETAILS_UK_ROOT,
+  },
+} = INSURANCE_ROUTES;
+
+const baseUrl = Cypress.config('baseUrl');
+
+context('Insurance - Policy - Loss Payee Financial details UK - Financial address - Validation', () => {
```
Comment (Zainzzkk) [inline e2e-tests/insurance/cypress/e2e/journeys/policy/loss-payee-financial-details-uk/validation/financial-address-validation.spec.js:13]: D should be capital.  Not a blocker

File: src/ui/server/controllers/insurance/policy/loss-payee-financial-details-international/validation/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import validation from '.';
+import validationRules from './rules';
+import FIELD_IDS from '../../../../../constants/field-ids/insurance/policy';
+import combineValidationRules from '../../../../../helpers/combine-validation-rules';
+
+const {
+  LOSS_PAYEE_FINANCIAL_UK: { ACCOUNT_NUMBER },
```
Comment (Zainzzkk) [inline src/ui/server/controllers/insurance/policy/loss-payee-financial-details-international/validation/index.test.ts:7]: need `LOSS_PAYEE_FINANCIAL_INTERNATIONAL` field IDs here

File: e2e-tests/insurance/cypress/e2e/journeys/policy/loss-payee-financial-details-uk/validation/financial-address-validation.spec.js (hunk 1 above)
Comment (ttbarnes) [reply e2e-tests/insurance/cypress/e2e/journeys/policy/loss-payee-financial-details-uk/validation/financial-address-validation.spec.js]: Fixed

File: src/ui/server/controllers/insurance/policy/loss-payee-financial-details-international/validation/index.test.ts (hunk 2 above)
Comment (ttbarnes) [reply src/ui/server/controllers/insurance/policy/loss-payee-financial-details-international/validation/index.test.ts]: Actually don't need and field IDs in such validation tests with an empty body - removed

**Review by Zainzzkk — APPROVED** [review]
